src/tech/v3/dataset/impl/column_index_structure.clj

```diff
@@ -54,7 +54,7 @@
                                                ^LinkedHashMap (LinkedHashMap.))]
         (if as-index-structure
           picked-map
-          (ListPersistentVector. (first (.values ^Map picked-map))))))))
+          (ListPersistentVector. (reduce concat [] (.values picked-map))))))))
```

Comment:
What if you do (.addAll (ArrayList.))?  And going further down that pathway what if you do (.addAll (dtype/make-list :int64))?

Reply:
Thank you @cnuernber! I will try these variations.

Reply:
Hey Chris, here's some times for the different approaches. I might not have translated what you mean quite right. The reason we have to loop over the elements of `(.values ...)` is that each value is a list of index positions.

```clj
(c/quick-bench
 (-> xidx
     (.subMap 100000 true 200000 true)
     (.values)
     (->> (reduce into (tech.v3.datatype.ListPersistentVector. [])))))
;; Execution time mean : 23.044314 sec

(c/quick-bench
 (-> xidx
     (.subMap 100000 true 200000 true)

     (.values)
     (->> (reduce concat []))
     (tech.v3.datatype.ListPersistentVector.)
     ))
;; Execution time mean : 2.879188 ms

(c/quick-bench
 (let [lst (java.util.ArrayList.)
       values (-> xidx
                  (.subMap 100000 true 200000 true)
                  (.values))]
   (doall (map #(.addAll lst %) values))
   lst))
;; Execution time mean : 11.996928 ms

(c/quick-bench
 (let [lst (tech.v3.datatype/make-list :int64)
       values (-> xidx
                  (.subMap 100000 true 200000 true)
                  (.values))]
   (doall (map #(.addAll lst %) values))))
;; Execution time mean : 76.205408 ms
```

Reply:
I worked with the int64 pathway and ended up with:

```clojure
  (c/quick-bench
   (let [values (-> xidx
                    (.subMap 100000 true 200000 true)
                    (.values))
         lst (dt/make-list :int64)]
     (pfor/doiter lst-data values
                  (pfor/doiter idx lst-data
                               (.addLong lst (unchecked-long idx))))
     lst))
```

Either the arraylist pathway or the above tech pathway will get the job done faster with the arraylist pathway being faster and the tech pathway being potentially more memory efficient.

Reply:
The reason the tech list pathway's addAll is slower is because it is optimized for the case where you are adding two large lists together in which case it falls back to dtype/copy.  For this case where each list is a persistent vector of length 1 that pathway is very inefficient.

`pfor` above is tech.v3.parallel.for.

Reply:
Excellent, @cnuernber. Thank you for the generous explanations. That `concat` problem is really interesting, I learned a lot. 

I will integrate your suggestion.